### PR TITLE
chore: Update clients

### PR DESCRIPTION
**Pull request by coszio**
Updates clients with the latest dev changes.

For grpc I could not install `grpcio==1.48.2`, so I updated it with `grpcio==1.49.1`

New stuff:
- facets
- distance matrix
- strict mode config
- misc

### CODE REVIEW

**Comment by joein** [comment]
@coszio 

what was the problem with grpcio?
1.48.2 works fine for me

using 1.49.1 brings type checker issues :(

**Comment by coszio** [comment]
I believe what we need is to pin the `grpcio-tools` version to 1.48.2, but not necessarily for `grpcio`. 

In my computer trying to build `grpcio==1.48.2` fails every time, by relaxing this one, I could generate correctly, as far as I can tell

File: tools/generate_grpc_client.sh [hunk 1]
```diff
@@ -4,8 +4,8 @@ set -e
 
 PROJECT_ROOT="$(pwd)/$(dirname "$0")/../"
 
-pip install grpcio==1.48.2
-pip install grpcio-tools==1.48.2
+pip install "grpcio>=1.48.2,<2.0.0"
+pip install grpcio-tools==1.48.2 # the expected python version to use is 1.10.10
```
Comment (joein) [inline tools/generate_grpc_client.sh:8]:
```suggestion
pip install grpcio-tools==1.48.2 # the expected python version to use is 3.10.10
```

**Review by joein — APPROVED** [review]

**Review by generall — CHANGES_REQUESTED** [review]
Please generate grpc with original version of `grpcio`

Do not change the version.

**Review by generall — APPROVED** [review]

**Pull request merged**